File: 002/main.c
#include <stdio.h>

int main(void) {
  long long a = 1, b = 2, sum = 2;

  while (b <= 4e6) {
    long long x = a + b;
    sum += x * !(x % 2);
    a = b;
    b = x;
  }

  printf("%lld\n", sum);

  return 0;
}

File: 002/main.py
from itertools import takewhile

def fibs():
    a, b = 0, 1
    while True:
        a, b = b, a + b
        yield b

if __name__ == "__main__":
    print(sum(takewhile(lambda x: x <= int(4e6), filter(lambda x: x % 2 == 0, fibs()))))
